homeassistant/components/yale_smart_alarm/__init__.py

```diff
@@ -1,22 +1,25 @@
 """The yale_smart_alarm component."""
 from __future__ import annotations
 
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
 
 from .const import DOMAIN, LOGGER, PLATFORMS
 from .coordinator import YaleDataUpdateCoordinator
 
 
-async def async_setup_entry(hass, entry):
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
```

Comment:
The return value should be a boolean.
```suggestion
async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```